compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -3097,12 +3087,9 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
             }
         };
 
-        let error = match error {
-            Ok(()) => {
-                self.report_elided_lifetime_in_ty(lifetime_refs);
-                return;
-            }
-            Err(error) => error,
+        let Err(error) = error else {
+            self.report_elided_lifetime_in_ty(lifetime_refs);
+            return;
```

Comment:
This else block corresponds to the "success" case, which feels a bit weird.

Reply:
Personally, that's why I like `let else` so much: it is agnostic whether it's looking for "success" or for "failure". But I'll remove the change.